.github/workflows/link-checker.yml

```diff
@@ -0,0 +1,56 @@
+name: Link Checker
+
+on:
+  pull_request:
+    branches:
+      - docs-staging
+      - master
+
+jobs:
+  test:
+    name: Check for broken links
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Fetch Git refs before link check to avoid rate limiting
+      - name: List changed files
+        uses: tj-actions/changed-files@v35
+
+      # Workaround for Lychee not recognizing MD links without extensions as valid
+      - name: Create symlinks
+        run: |
+          find docs -name "*.md" -type f | while read f; do
+            ln -s $(basename $f) ${f%.*}
+          done
+
+      # Disable cache until https://github.com/lycheeverse/lychee/issues/840 is fixed
+      # - name: Restore lychee cache
+      #   uses: actions/cache@v3
+      #   with:
+      #     path: .lycheecache
+      #     key: cache-lychee-${{ github.ref_name }}
+      #     restore-keys: cache-lychee-
+
+      - name: Run lychee
+        uses: lycheeverse/lychee-action@v1.5.4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          # args: "--accept 401,429 --cache --max-cache-age 1d --max-concurrency 64 --no-progress --verbose 'docs/**/*.md'"
+          args: "--accept 401,429 --max-concurrency 64 --no-progress --verbose 'docs/**/*.md'"
+
+      - name: List all files with errors
+        run: cat lychee/out.md | grep "# Errors in" | cut -f4- -d' ' > lychee/files.txt
+
+      - name: List changed files with errors
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+        with:
+          files_from_source_file: lychee/files.txt
+
+      - name: Fail if new errors found
+        if: steps.changed-files.outputs.any_changed == 'true'
```

Comment:
Curious if this fails when reducing the number of broken links? 😋 
say that someone only addresses 1 broken link... 
will this consider the `changed-files.outputs.any_changed == true` ?

Reply:
After consulting with the docs squad, decided to simplify the logic so the PR fails for any broken links found in the entire repo.